### DISCUSSION TITLE
Update Slice State After Manual Cache Update

### DISCUSSION THREAD

**Discussion by User6531** [reduxjs/redux-toolkit#4403] [Q&A] [answered]
Hi everyone,

I'm facing an issue where the state of a `slice` in my application isn't updating correctly after I manually update the cache. The slice contains a property dependent on the number of users, recalculated through an `addMatcher` in `extraReducers`. However, when I update the cache manually, this doesn't trigger the `matcher`, leaving the property unchanged.

Any suggestions on how to ensure the property recalculates after manual cache updates would be appreciated.

Thanks.

```tsx
const userSlice = createSlice({
  name: "userSlice",
  initialState: { usersAmount: 0 },
  reducers: {},
  extraReducers: (builder) => {
    builder.addMatcher(
      usersApi.endpoints.getUsers.matchFulfilled,
      (state, { payload }) => {
        state.usersAmount = payload.length;
      },
    );
  },
});

const usersApi = mainApi.injectEndpoints({
  endpoints: (builder) => ({
    getUsers: builder.query<TUser[], void>({
      query: () => `${usersBaseUrl}/users`,
    }),

    deleteUser: builder.mutation<void, string>({
      query: (userId) => ({
        url: `${usersBaseUrl}/users/${userId}`,
        method: "DELETE",
      }),
  
      async onQueryStarted(userId, { dispatch, queryFulfilled }) {
        try {
          await queryFulfilled;
  
          dispatch(
            usersApi.util.updateQueryData("getUsers", undefined, (prevData) =>
              prevData.filter((user) => user.id !== userId),
            ),
          );
        } catch {}
      },
    })
  }),
});


```

**Comment by EskiMojo14** [comment]
are you editing userAmount anywhere else, or is it purely derived from the query data?

**Reply by phryneas** [reply]
Yeah, just from this snippet this looks like a value that you definitely should not be holding in a Redux store on it's own.

**Reply by User6531** [reply]
@EskiMojo14 @phryneas  

My code is just a simplified example.
`userAmount` does not change anywhere else; it is purely derived from the query data. 
The reason why I hold it in the redux store is that I need this value inside my header component which is permanently on page. 
If I try to use the query hook inside a header, the data will always be in use and never unsubscribe, which causes the cache to never clear and refetch again. 
I want to keep the logic when the data will invalidate 60 seconds after unsubscribe.

**Reply by phryneas** [reply]
To me that sounds like you should set the `refetchOnMountOrArgChange` option to `useQuery` in your non-header component to 60 seconds instead.

**Reply by User6531** [reply]
Yea, that what I have right now, it works, but I have about 5 non-header components that use `useQuery`, and it could become more. Because of that, I am thinking of changing the logic a bit.

**Comment by User6531** [comment]
I assume that there is no possibility to auto trigger `addMatcher` in `extraReducers` after manual cache updating.

**Reply by EskiMojo14** [reply, marked as answer]
there is no action dispatched that would be useful to you, no - the only action actually dispatched contains JSON patches of your changes that are applied in the reducer itself.
https://github.com/reduxjs/redux-toolkit/blob/master/packages/toolkit/src/query/core/buildSlice.ts#L130
https://github.com/reduxjs/redux-toolkit/blob/master/packages/toolkit/src/query/core/buildThunks.ts#L251

**Reply by User6531** [reply]
thank you for your reply, much appreciated
